jupiterone/client.py

```diff
@@ -55,6 +55,8 @@ def __init__(self, account: str = None, token: str = None, url: str = DEFAULT_UR
             'Authorization': 'Bearer {}'.format(self.token),
             'LifeOmic-Account': self.account
         }
+        self.headers.update(kwargs.get('headers', {}))
+        self.verify = kwargs.get('verify', True)
```

Comment:
Nah we shouldn't support turning TLS verification off. We _can_ support providing a custom certificate chain though. So instead of a boolean, this could be used to point to a custom chain:

```
requests.get('https://github.com', verify='/path/to/certfile')
```

We should make sure that's defined as a non-zero length path that exists, too, else fail.

Reply:
Yes, that is how I was testing it. But the requests documentation says that the default value for it is True. I don't know if further validations here are worth, or that would be more requests' responsibility, as here we just forward the value if it comes. If was doing some tests with the bundle like the one that follows:

```python
#!/usr/bin/env python3

import os
import tempfile
import subprocess

from contextlib import contextmanager
from jupiterone import JupiterOneClient

def run_command(command):
  result = subprocess.run(
        command,
        shell=True,
        stdout=subprocess.PIPE,
        stderr=subprocess.PIPE,
        universal_newlines=True)
  if result.returncode == 0:
    return result.stdout
  else:
    raise Exception(f'Error obtaining certificate: {result.stderr}')

@contextmanager
def make_certs_dir():
  with tempfile.TemporaryDirectory() as tmpdir:
    for i, cert in enumerate([
      run_command(cmd) for cmd in [
        'security find-certificate -c "Okta Inc Root CA" -a -p',
        'security find-certificate -c "Okta Inc Forward Trust CA" -a -p'
      ]
    ]):
      with open(os.path.join(tmpdir, f'{i}.pem'), 'w') as f:
        f.write(cert)

    run_command(f'c_rehash {tmpdir}')

    yield tmpdir


def main():
  with make_certs_dir() as tmpdir:
    account = os.environ.get('J1_ACCOUNT_ID')
    token = os.environ.get('J1_TOKEN')
    j1 = JupiterOneClient(account, token, verify=tmpdir)

    QUERY = 'FIND Host'
    query_result = j1.query_v1(QUERY, limit=5)
    print(query_result)

if __name__ == "__main__":
  main()
```

Reply:
Ah I see, yes that makes sense. I'd still vote we do some validation on that arg we're processing, or at least a docstring, to ensure it doesn't suggest we're at all condoning users skip TLS altogether.